javascript/ql/lib/semmle/javascript/frameworks/Express.qll

```diff
@@ -1071,4 +1072,29 @@ module Express {
 
     override predicate definitelyResumesDispatch() { none() }
   }
+
+  /**
+   * An express route setup configured with the `cors` package.
+   */
+  class CorsConfiguration extends DataFlow::MethodCallNode {
+    CorsConfiguration() {
+      exists(Express::RouteSetup setup | this = setup |
+        setup.isUseCall() and setup.getArgument(0) instanceof Cors::Cors
+        or
+        not setup.isUseCall() and setup.getAnArgument() instanceof Cors::Cors
+      )
+    }
+
+    /** Gets the cors argument */
+    Cors::Cors getArgument() { result = this.getArgument(0) }
+
+    /** Gets the options used to configure `cors`. */
+    DataFlow::Node getCorsArgument() { result = this.getArgument().getOptionsArgument() }
+
+    /** Holds if cors is using its default configuration. */
+    predicate isDefault() { this.getArgument().isDefault() }
+
+    /** Gets the `origin` option that the call to `cors` is configured with. */
+    DataFlow::Node getOrigin() { result = this.getArgument().getOrigin() }
+  }
```

Comment:
```suggestion
  class CorsConfiguration extends DataFlow::MethodCallNode {
    Cors::Cors corsConfig;

    CorsConfiguration() {
      exists(Express::RouteSetup setup | this = setup |
        if setup.isUseCall()
        then corsConfig = setup.getArgument(0)
        else corsConfig = setup.getArgument(any(int i | i > 0))
      )
    }

    /** Gets the expression that configures `cors` on this route setup. */
    Cors::Cors getCorsConfiguration() { result = corsConfig }
  }
```

This adds `corsConfig` as a field, so the (renamed) `getCorsConfiguration` predicate no longer assumes that only the first argument can be the configuration of `cors`. 

Also, I deleted some of the predicates, as they are not really needed.